### DISCUSSION TITLE
Dealing with the outputs of MCL

### DISCUSSION THREAD

**Discussion by AlexAG1996** [micans/mcl#7] [General]
Hi all, 
I am a beginner to using MCL, so I have generated the output file in the ABC format. I am wondering if there is a way to extract the clusters individually and place them in separate files. I am also wondering if there a way to limit the number of clusters that are generated. 

Kind regards, 
Alex

**Comment by micans** [comment]
Hi Alex,

I assume you are working on a linux or unix-type system. The mcl output orders clusters from large to small; I assume you are interested in the N largest clusters. E.g. if N=10 you can do

```
head -n 10 MCL-OUTPUT-FILE | split -l 1 - foo.
```

This will create files `foo.aa`, `foo.ab`, `foo.ac` and so on. If you want numeric suffixes:

```
head -n 10 MCL-OUTPUT-FILE | split -l 1 - foo. --numeric-suffixes=1 --verbose
```

Change `foo` to some other string to pick the prefix you want. `-l 1` indicates split will split line-by-line, and the single dash `-` after that indicates that it reads from STDIN (which is what it gets sent by the `head` program). Hope this is what you need, let me know if you have further questions.

**Reply by AlexAG1996** [reply]
Thanks Micans, 
just on the off chance do you know how to extract the names contained within each cluster and place them into a format that can be used as a dictionary please. 

Kind regards, 
Alex

**Comment by micans** [comment]
Hi,

That can be done in all programming languages, I imagine a sequence of file read, splitting a line, and setting dictionary keys is required. Each of these should be fairly easy to find examples of in your favourite language.

Good luck!
Stijn

**Reply by AlexAG1996** [reply]
thanks Stijn 🤠
